woocommerce/woocommerce-android
# [Dynamic Dashboard] Fix issue of not updating the orders count in some scenarios

### Description
In some cases, and given the fact that order status options are fetched only on app launch and when switching sites, it could happen that the app shows the performance card for stores with no orders or vice versa.

This PR fixes this by making two changes:
1. It updates the logic of `ObserveProcessingOrdersCount` to always use the order status API for infering the number of processing orders, instead of using the `/wc-analytics/reports/totals` endpoint, this update will allow us to refresh the cache whenever any changes happen.
2. It updates the logic of `ObserveSiteOrdersState` to observe the changes of order status options instead of retrieving a single snapshot.

With these two changes, the dashboard will always observe the updated state.

### Testing instructions
1. Use a store with a single order.
2. Open the app.
3. Notice the performance card is shown.
4. Switch to the order list screen.
5. Delete the order.
6. Go back to the dashboard screen.
7. Notice the performance card was disabled (there could be a small delay for this, and it's because of the delay of the `undo` action that we use before trashing the order).
8. Repeat the test with "adding a new order"
9. Notice the performance card was enabled.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1657201/75de6a5a-537d-4fd6-878c-22bac62c5e47
